### PR TITLE
Add Javascript to Wagtail Admin to Disable Fig Content Page Section_ID field

### PR DESCRIPTION
**Pull request by cooldragontattoo**
The section_id in the Wagtail Admin of the Fig_Content_Page is not to be interacted with by the user.  Since Wagtail lacks internal functionality that makes textfields readonly, the section_id is rendered readonly using custom frontend javascript.

## Additions

- added a FigSectionAdapter
- added a custom javascript for the FigSectionAdapter
- added a js directory in `templates/wagtailadmin`

## How to test this PR

1. pull branch and make a FigContentPage in the Wagtail admin
2. discover the wonder that is a disabled section_id in the FigSection, FigSubSection, FigSub3Section

## Screenshots
|before|after|
|-|-|
|Screen Shot 2022-06-08 at 10 15 29 AM|Screen Shot 2022-06-08 at 10 13 59 AM|

## Checklist

- [ ] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: cfgov/templates/wagtailadmin/js/fig-section-block.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class FigSectionDefintion extends window.wagtailStreamField.blocks.StructBlockDefinition {
+    render(placeholder, prefix, initialState, initialError) {
+        const block = super.render(placeholder, prefix, initialState, initialError);
+
+        console.log(placeholder);
+        console.log(prefix);
+
```
Comment (chosak) [inline cfgov/templates/wagtailadmin/js/fig-section-block.js:7]:
```suggestion
```

**Review by chosak — COMMENTED** [review]
One minor removal of console logging.

Besides that, I suggest moving the FIG-related StreamField block types to a new file at filing_instruction_guide/blocks.py, and then modifying the page model to pull those in from there, instead of adding them to the global organisms.py. The atomic_elements modules have become something of a dumping ground for all kinds of blocks, even those that are specifically tailored to a single app like this one.

Similarly, you could then move the JS file under a new filing_instruction_guide/static/js/ folder, which would consolidate all FIG-related logic into the FIG app.

Functionally the above two changes shouldn't impact how it works, but would be a nice cleanup to keep FIG code together.

**Comment by willbarton** [comment]
> Besides that, I suggest moving the FIG-related StreamField block types to a new file at filing_instruction_guide/blocks.py, and then modifying the page model to pull those in from there, instead of adding them to the global organisms.py. The atomic_elements modules have become something of a dumping ground for all kinds of blocks, even those that are specifically tailored to a single app like this one.

I would second this. 

One thing we've struggled with in the past with Wagtail upgrades and migration squashing is circular dependencies between apps (i.e. we used to have Ask CFPB blocks defined here, such that `v1` depended on `ask_cfpb`, while `ask_cfpb` depended on `v1`). The fact that they're part of the same codebase makes it harder to enforce, but generally, something should only be in `v1` if it is common to the entire CMS, and anything specific to apps built on the CMS should be defined within that app. 

So, I agree, the FIG-specific blocks should probably live in a `filing_instruction_guide/blocks.py` file.

**Comment by cooldragontattoo** [comment]
@chosak, you had mentioned that the there was another way to connect the js to the blocks, should they live in `filing_instruction_guide/blocks.py`.  would it be optimal to also move the js too?

**Comment by cooldragontattoo** [comment]
great--i moved the file so this should be ready for approval @schbetsy @chosak

File: cfgov/filing_instruction_guide/static/js/fig-section-block.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class FigSectionDefintion extends window.wagtailStreamField.blocks.StructBlockDefinition {
```
Comment (chosak) [inline cfgov/filing_instruction_guide/static/js/fig-section-block.js:1]:
```suggestion
class FigSectionDefinition extends window.wagtailStreamField.blocks.StructBlockDefinition {
```

File: cfgov/filing_instruction_guide/static/js/fig-section-block.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+        const block = super.render(placeholder, prefix, initialState, initialError);
+
+        const sectionID = $(document).find('#' + prefix + '-section_id');
+        sectionID.prop('disabled', true);
+
+        return block;
+    }
+}
+
+window.telepath.register('filing_instruction_guide.blocks.FigSection', FigSectionDefintion);
```
Comment (chosak) [inline cfgov/filing_instruction_guide/static/js/fig-section-block.js:12]:
```suggestion
window.telepath.register('filing_instruction_guide.blocks.FigSection', FigSectionDefinition);
```

**Review by schbetsy — APPROVED** [review]
I'm blundering into this PR at the last minute, but this looks great to me. Thanks @chosak and @willbarton for collaborating.

**Pull request merged**